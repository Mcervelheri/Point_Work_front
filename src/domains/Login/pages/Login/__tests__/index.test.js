import React from 'react';

import { render, screen } from '@testing-library/react';
import { createMemoryHistory } from 'history';
import { Provider } from 'react-redux';
import { Router } from 'react-router-dom';
import { create } from 'react-test-renderer';
import configureMockStore from 'redux-mock-store';

import LoginPage from '../index';

const mockStore = configureMockStore();
const store = mockStore({});
const history = createMemoryHistory({
    initialEntries: [''],
});
describe('Testa tela login \n', () => {

    test('<LoginPage> deve renderizar corretamente', () => {

        const loginPage = create(
            <Router history={history}>
                <Provider store={store}>
                    <LoginPage />
                </Provider>
            </Router>,
        ).toJSON();

        expect(loginPage).toMatchSnapshot();
    });

    test('Deve conter a logotipo da AL5 Bank na tela LoginPage ', () => {
        render(
            <Router history={history}>
                <Provider store={store}>
                    <LoginPage />
                </Provider>
            </Router>,
        );
        const logo = screen.getByAltText(/Logotipo AL5 Bank/i);
        expect(logo).toBeInTheDocument();
    });

    test('Deve conter o título Login na tela LoginPage ', () => {
        render(
            <Router history={history}>
                <Provider store={store}>
                    <LoginPage />
                </Provider>
            </Router>,
        );
        const loginTitle = screen.getByText(/Login/i);
        expect(loginTitle).toBeInTheDocument();
    });

    test('Deve conter o input de email na tela LoginPage ', () => {
        render(
            <Router history={history}>
                <Provider store={store}>
                    <LoginPage />
                </Provider>
            </Router>,
        );
        const emailInput = screen.getByPlaceholderText('Email');
        expect(emailInput).toBeInTheDocument();
    });

    test('Deve conter o input de senha na tela LoginPage ', () => {
        render(
            <Router history={history}>
                <Provider store={store}>
                    <LoginPage />
                </Provider>
            </Router>,
        );
        const passwordInput = screen.getByPlaceholderText('Senha');
        expect(passwordInput).toBeInTheDocument();
    });

    test('Deve conter checkbox de lembrar credenciais na tela LoginPage ', () => {
        render(
            <Router history={history}>
                <Provider store={store}>
                    <LoginPage />
                </Provider>
            </Router>,
        );
        const rememberButton = screen.getByRole('checkbox', { name: 'Lembrar' });
        expect(rememberButton).toBeInTheDocument();
    });

    test('Deve conter o botão de recuperar senha na tela LoginPage ', () => {
        render(
            <Router history={history}>
                <Provider store={store}>
                    <LoginPage />
                </Provider>
            </Router>,
        );
        const recoverPassword = screen.getByRole('button', { name: 'Esqueceu sua senha?' });
        expect(recoverPassword).toBeInTheDocument();
    });

    test('Deve conter o botão Acessar na tela LoginPage', () => {
        render(
            <Router history={history}>
                <Provider store={store}>
                    <LoginPage />
                </Provider>
            </Router>,
        );
        const accessButton = screen.getByRole('button', { name: /Acessar/i });
        expect(accessButton).toBeInTheDocument();
    });

    test('Deve conter o texto AL5 Bank no rodapé da tela LoginPage', () => {
        render(
            <Router history={history}>
                <Provider store={store}>
                    <LoginPage />
                </Provider>
            </Router>,
        );
        const footerTitle = screen.getByText(/AL5 Bank/i);
        expect(footerTitle).toBeInTheDocument();
    });

});
