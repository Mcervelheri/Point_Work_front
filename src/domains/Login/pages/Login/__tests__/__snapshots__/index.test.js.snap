// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testa tela login 
 <LoginPage> deve renderizar corretamente 1`] = `
<div
  className="container"
>
  <img
    alt="Logotipo AL5 Bank"
    src="logo-al5-dark.svg"
  />
  <p>
    Login
  </p>
  <div
    className="formContainer"
  >
    <div
      className="ant-form ant-form-vertical"
      onReset={[Function]}
      onSubmit={[Function]}
    >
      <form
        noValidate={true}
        onSubmit={[Function]}
      >
        <div
          className="ant-form-item"
        >
          <div
            className="ant-row ant-form-item-row"
            style={Object {}}
          >
            <div
              className="ant-col ant-form-item-control"
              style={Object {}}
            >
              <div
                className="ant-form-item-control-input"
              >
                <div
                  className="ant-form-item-control-input-content"
                >
                  <span
                    className="ant-input-affix-wrapper"
                    onClick={[Function]}
                  >
                    <span
                      className="ant-input-prefix"
                    >
                      <span
                        aria-label="user"
                        className="anticon anticon-user prefixIcon"
                        role="img"
                      >
                        <svg
                          aria-hidden="true"
                          data-icon="user"
                          fill="currentColor"
                          focusable="false"
                          height="1em"
                          viewBox="64 64 896 896"
                          width="1em"
                        >
                          <path
                            d="M858.5 763.6a374 374 0 00-80.6-119.5 375.63 375.63 0 00-119.5-80.6c-.4-.2-.8-.3-1.2-.5C719.5 518 760 444.7 760 362c0-137-111-248-248-248S264 225 264 362c0 82.7 40.5 156 102.8 201.1-.4.2-.8.3-1.2.5-44.8 18.9-85 46-119.5 80.6a375.63 375.63 0 00-80.6 119.5A371.7 371.7 0 00136 901.8a8 8 0 008 8.2h60c4.4 0 7.9-3.5 8-7.8 2-77.2 33-149.5 87.8-204.3 56.7-56.7 132-87.9 212.2-87.9s155.5 31.2 212.2 87.9C779 752.7 810 825 812 902.2c.1 4.4 3.6 7.8 8 7.8h60a8 8 0 008-8.2c-1-47.8-10.9-94.3-29.5-138.2zM512 534c-45.9 0-89.1-17.9-121.6-50.4S340 407.9 340 362c0-45.9 17.9-89.1 50.4-121.6S466.1 190 512 190s89.1 17.9 121.6 50.4S684 316.1 684 362c0 45.9-17.9 89.1-50.4 121.6S557.9 534 512 534z"
                          />
                        </svg>
                      </span>
                    </span>
                    <input
                      className="ant-input"
                      hidden={null}
                      id="username"
                      name="username"
                      onBlur={[Function]}
                      onChange={[Function]}
                      onFocus={[Function]}
                      onKeyDown={[Function]}
                      placeholder="Email"
                      required={true}
                      style={null}
                      type="text"
                      value=""
                    />
                  </span>
                </div>
              </div>
            </div>
          </div>
        </div>
        <div
          className="ant-form-item"
        >
          <div
            className="ant-row ant-form-item-row"
            style={Object {}}
          >
            <div
              className="ant-col ant-form-item-control"
              style={Object {}}
            >
              <div
                className="ant-form-item-control-input"
              >
                <div
                  className="ant-form-item-control-input-content"
                >
                  <span
                    className="ant-input-affix-wrapper"
                    onClick={[Function]}
                  >
                    <span
                      className="ant-input-prefix"
                    >
                      <span
                        aria-label="lock"
                        className="anticon anticon-lock prefixIcon"
                        role="img"
                      >
                        <svg
                          aria-hidden="true"
                          data-icon="lock"
                          fill="currentColor"
                          focusable="false"
                          height="1em"
                          viewBox="64 64 896 896"
                          width="1em"
                        >
                          <path
                            d="M832 464h-68V240c0-70.7-57.3-128-128-128H388c-70.7 0-128 57.3-128 128v224h-68c-17.7 0-32 14.3-32 32v384c0 17.7 14.3 32 32 32h640c17.7 0 32-14.3 32-32V496c0-17.7-14.3-32-32-32zM332 240c0-30.9 25.1-56 56-56h248c30.9 0 56 25.1 56 56v224H332V240zm460 600H232V536h560v304zM484 701v53c0 4.4 3.6 8 8 8h40c4.4 0 8-3.6 8-8v-53a48.01 48.01 0 10-56 0z"
                          />
                        </svg>
                      </span>
                    </span>
                    <input
                      className="ant-input"
                      hidden={null}
                      name="password"
                      onBlur={[Function]}
                      onChange={[Function]}
                      onFocus={[Function]}
                      onKeyDown={[Function]}
                      placeholder="Senha"
                      required={true}
                      style={null}
                      type="password"
                      value=""
                    />
                  </span>
                </div>
              </div>
            </div>
          </div>
        </div>
        <div
          className="checkboxContainer"
        >
          <label
            className="ant-checkbox-wrapper"
          >
            <span
              className="ant-checkbox"
              style={Object {}}
            >
              <input
                checked={false}
                className="ant-checkbox-input"
                disabled={false}
                onBlur={[Function]}
                onChange={[Function]}
                onFocus={[Function]}
                onKeyDown={[Function]}
                onKeyPress={[Function]}
                onKeyUp={[Function]}
                type="checkbox"
              />
              <span
                className="ant-checkbox-inner"
              />
            </span>
            <span>
              Lembrar
            </span>
          </label>
          <button
            className="ant-btn ant-btn-link ant-btn-lg link"
            disabled={false}
            onClick={[Function]}
            type="button"
          >
            <span>
              Esqueceu sua senha?
            </span>
          </button>
        </div>
        <button
          className="ant-btn ant-btn-primary ant-btn-block"
          disabled={false}
          onClick={[Function]}
          type="submit"
        >
          <span>
            Acessar
          </span>
        </button>
      </form>
    </div>
  </div>
  <div
    className="footerTitle"
  >
    <p>
      AL5 Bank
    </p>
  </div>
</div>
`;
